### ISSUE TITLE
On Zen 2, it's seemingly impossible to submit traces to Pernosco

### ISSUE THREAD

**Issue by lf-** [Pernosco/pernosco-submit#13]
I try to make a trace, but Pernosco rejects it:
```
$ ~/dev/pernosco-submit/pernosco-submit upload --dry-run /tmp/vv ~/.local/share/rr/cargo-9/ ~/dev/cargo
CPU feature SHA extensions enabled when recording trace, but Pernosco does not support SHA extensions yet.
Re-record with `rr record --disable-cpuid-features-ext 0xfc230000,0x2c42,0xc`.
```

I then try the suggested flags, which fail because AMD appears to not support CPUID faulting (based on both `cpuid_fault` not being in /proc/cpuinfo and reading the linux source code which only has that flag being a thing on Intel CPUs):

```
$ (cd kern; rr record --disable-cpuid-features-ext 0xfc230000,0x2c42,0xc -- ~/dev/cargo2/target/debug/cargo b --release)
rr: Saving execution to trace directory `/home/jade/.local/share/rr/cargo-10'.
[FATAL ../src/RecordSession.cc:2264:RecordSession() errno: EOPNOTSUPP] CPUID faulting required to disable CPUID features
=== Start rr backtrace:
rr(_ZN2rr13dump_rr_stackEv+0x42)[0x559ed14d8002]
rr(_ZN2rr15notifying_abortEv+0x50)[0x559ed14d80a0]
rr(_ZN2rr12FatalOstreamD1Ev+0x53)[0x559ed13d0a23]
rr(_ZN2rr13RecordSessionC2ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt6vectorIS6_SaIS6_EESD_RKNS_20DisableCPUIDFeaturesENS0_16SyscallBufferingEiNS_7BindCPUES8_PKNS_9TraceUuidEbb+0x21b)[0x559ed1401f7b]
rr(_ZN2rr13RecordSession6createERKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS7_EESB_RKNS_20DisableCPUIDFeaturesENS0_16SyscallBufferingEhNS_7BindCPUERKS7_PKNS_9TraceUuidEbbb+0xb3b)[0x559ed1402cfb]
rr(_ZN2rr13RecordCommand3runERSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS7_EE+0x878)[0x559ed13f4998]
rr(main+0x16f)[0x559ed13641ef]
/usr/lib/libc.so.6(__libc_start_main+0xd5)[0x7fdcfa432b25]
rr(_start+0x2e)[0x559ed136434e]
=== End rr backtrace
[1]    24113 abort (core dumped)  ( cd kern; rr record --disable-cpuid-features-ext 0xfc230000,0x2c42,0xc --  b)
```

relevant bits of `/proc/cpuinfo`:

```
processor       : 0
vendor_id       : AuthenticAMD
cpu family      : 23
model           : 113
model name      : AMD Ryzen 9 3900X 12-Core Processor
stepping        : 0
microcode       : 0x8701021
cpu MHz         : 3800.000
cache size      : 512 KB
physical id     : 0
siblings        : 24
core id         : 0
cpu cores       : 12
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 16
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext
 fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq monitor ssse3 fma cx16 sse
4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs
 skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate sme ssbd mba sev ibpb stibp vmmcall
 sev_es fsgsbase bmi1 avx2 smep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_l
lc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid
 decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif umip rdpid overflow_recov succor smca
bugs            : sysret_ss_attrs spectre_v1 spectre_v2 spec_store_bypass
bogomips        : 7603.41
TLB size        : 3072 4K pages
clflush size    : 64
cache_alignment : 64
address sizes   : 43 bits physical, 48 bits virtual
power management: ts ttp tm hwpstate cpb eff_freq_ro [13] [14]
```

**Comment by rocallahan** [comment]
You can resubmit with `--ignore-warnings` and that should work. Unless your application actually uses the SHA extensions, in which case we're out of luck :-(.

**Comment by lf-** [comment]
> You can resubmit with `--ignore-warnings` and that should work. Unless your application actually uses the SHA extensions, in which case we're out of luck :-(.

Oh cool, that works. I thought it was a hard warning. Maybe the guidance message needs to be changed for CPUs that don't support CPUID faults.

I also ran into another footgun more or less caused by `rr sources`: when I built rr from git, if I don't strip the `librrpreload.so`, `rr sources` includes the assembly sources from rr itself in there, which then causes `pernosco-submit` to try to grab the git repo for those, and that blew up  since it looked inside a git repo where `git status` emitted a warning (no permissions for some random unrelated files that were created in a chroot), triggering the `Unexpected line` exception in `pernosco-submit`.
